R/distributions.R

```diff
@@ -1,16 +1,15 @@
 ## Not in base R
-rmvnorm <- function(x, vcv) {
-  make_rmvnorm(vcv)(x)
+rmvnorm <- function(x, vcv, rng) {
+  make_rmvnorm(vcv)(x, rng)
 }
 
 
-##' @importFrom stats rnorm
 make_rmvnorm <- function(vcv) {
   n <- ncol(vcv)
   r <- chol(vcv, pivot = TRUE)
   r <- r[, order(attr(r, "pivot", exact = TRUE))]
-  function(x) {
-    x + drop(rnorm(n) %*% r)
+  function(x, rng) {
```

Comment:
hmm, I'm not sure on this, but I can see where you're coming from here.

The reason I don't want to change it is with this version we will write at the point of use:

```
rmvnorm(x, rng)
```

Later we'll probably move the multivariate normal support within the rng class, at which point we'd write


```
rng$rmvnorm(x)
```

so this preserves the connection with `rng` and makes it clear we have the same RNG generator.  

We might also use the transformation captured by the closure here with any number of rng objects, perhaps in parallel (this will become more obvious in #8, I think when there are many `rng` objects floating around).  The way we have it here we capture `vcv`, which will be the same everwhere, but *not* the `rng` which can be different.

Reply:
This makes sense to me, I think, especially if it's going to end up in the rng class anyway. I suppose you could make rng an optional parameter on `rmvnorm` and if it's not provided then use a default rng, to allow for simpler usage. But maybe better to make the client think about appropriate rng usage for the scenario.